error.go

```diff
@@ -16,3 +21,35 @@ type marshalableError struct {
 func (m *marshalableError) MarshalJSON() ([]byte, error) {
 	return json.Marshal(m.err.Error())
 }
+
+type causer interface {
+	Cause() error
+}
+
+type stackTracer interface {
+	StackTrace() errors.StackTrace
+}
+
+func extractStackTrace(err error) errors.StackTrace {
+	var tracer stackTracer
+	for {
```

Comment:
I'm not familiar with the `errors` package, but I see in the README:

> errors.Cause will recursively retrieve the topmost error which does not implement causer, which is assumed to be the original cause. 

Why this `for` loop is needed then?